web/src/bootstrap_typeahead.ts

```diff
@@ -280,6 +297,10 @@ export class Typeahead<ItemType extends string | object> {
 
     select(e?: JQuery.ClickEvent | JQuery.KeyUpEvent | JQuery.KeyDownEvent): this {
         const val = this.values.get(this.$menu.find(".active")[0]!);
+        // It's possible that we got here from pressing enter with nothing highlighted.
+        if (this.allowNoHighlight && val === undefined) {
+            return this.hide();
+        }
```

Comment:
I'd rather we just not get to `select` at all in this case, but there are a lot of calls to `select` and I don't want to miss any.